app/controllers/reports/dashboards/overviews_controller.rb

```diff
@@ -5,14 +5,68 @@
 ###
 #
 class Reports::Dashboards::OverviewsController < ApplicationController
+  include PjaxModalController
+
   before_action :require_can_view_reports!
+  before_action :set_report
 
   def index
+  end
+
+  def details
+    @section = sections.detect { |name| params[:section].to_sym == name }
+    @key = params[:key]
```

Comment:
Let's be extra careful and limit these to `data.keys.detect...` and whatever the equivalent for sub-keys would bey, maybe `data.values.keys?`  Then we can return @clients = nil or some such if we don't have valid keys.